SignalServiceKit/src/Network/API/TSNetworkManager.m

```diff
@@ -56,6 +59,13 @@ - (void)makeRequest:(TSRequest *)request
             failure:(void (^)(NSURLSessionDataTask *task, NSError *error))failureBlock
 {
     DDLogInfo(@"%@ Making request: %@", self.logTag, request);
+    if (!CurrentAppContext().isMainApp) {
+        if (![request isKindOfClass:[TSRecipientPrekeyRequest class]]
+            && ![request isKindOfClass:[TSSubmitMessageRequest class]]) {
+            // The SAE should only make requests directly related to message sending.
+            OWSFail(@"%@ Making request: %@", self.logTag, request);
```

Comment:
I'm hoping to add a lot of asserts to help identify unexpected behavior in the SAE.  The more broad ones (like this) don't `return` or otherwise abort.